stan/math/prim/prob/wiener4_lcdf.hpp

```diff
@@ -0,0 +1,762 @@
+ *
+ * @param x A scalar
+ * @return Log of Mill's ratio
+ */
+template <typename T_x>
+inline auto logMill(T_x&& x) noexcept {
+  if (x > 1.0e5) {
+    return -log(x);
+  }
+  const auto m = std_normal_lcdf(-x) - std_normal_log(x);
+  return m;
+}
+
+/**
+ * Calculate the probability term 'P' on log scale for distribution
+ *
+ * @param a The boundary separation
+ * @param v The drift rate
+ * @param w The relative starting point
+ * @return 'P' term
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto log_probability_distribution(const T_a& a, const T_v& v,
+                                         const T_w& w) noexcept {
+  using ret_t = return_type_t<T_a, T_w, T_v>;
+  auto nearly_one = ret_t(1.0 - 1.0e-6);
+  if (fabs(v) == 0.0) {
+    return ret_t(log1p(-w));
+  }
+  auto minus_two_va_one_minus_w = (-2.0 * v * a * (1.0 - w));
+  ret_t prob;
+  if (minus_two_va_one_minus_w < 0) {
+    const auto exp_arg = exp(minus_two_va_one_minus_w);
+    if (exp_arg >= nearly_one) {
+      return ret_t(log1p(-w));
+    }
+    auto two_vaw = 2 * v * a * w;
+    if (two_vaw > minus_two_va_one_minus_w) {
+      prob = log1p(-exp_arg) - log_diff_exp(two_vaw, minus_two_va_one_minus_w);
+    } else if (two_vaw < minus_two_va_one_minus_w) {
+      prob = log1p(-exp_arg) - log_diff_exp(minus_two_va_one_minus_w, two_vaw);
+    } else {
+      prob = log1p(-exp_arg) - NEGATIVE_INFTY;
+    }
+  } else {
+    const auto exp_arg = exp(-minus_two_va_one_minus_w);
+    if (exp_arg >= nearly_one)
+      return ret_t(log1p(-w));
+    prob = log1p(-exp_arg) - log1p(-exp(2 * v * a));
+  }
+  return prob;
+}
+
+/**
+ * Calculate the probability term 'P' on log scale for grad_a and grad_v
+ *
+ * @param a The boundary separation
+ * @param v The drift rate
+ * @param w The relative starting point
+ * @return 'P' term
+ */
+template <typename T_a, typename T_w, typename T_v>
+inline auto log_probability_GradAV(const T_a& a, const T_v& v,
+                                   const T_w& w) noexcept {
+  using ret_t = return_type_t<T_a, T_w, T_v>;
+  auto nearly_one = ret_t(1.0 - 1.0e-6);
+  if (fabs(v) == 0.0) {
+    return ret_t(-w);
+  }
+  nearly_one = ret_t(1.0 - 1.1 * 1.0e-5);
+  ret_t prob;
+  if (v < 0) {
+    const auto two_va_one_minus_w = (2.0 * v * a * (1.0 - w));
+    const auto two_avw = 2 * a * v * w;
+    const auto two_av = 2 * a * v;
+    const auto exp_two_va_one_minus_w = exp(two_va_one_minus_w);
+    const auto exp_two_avw = exp(two_avw);
+    const auto exp_two_av = exp(two_av);
+    if (((exp_two_va_one_minus_w >= nearly_one) || (exp_two_avw >= nearly_one))
+        || (exp_two_av >= nearly_one)) {
+      return ret_t(-w);
+    }
+    prob = LOG_TWO + two_va_one_minus_w - log1p(-exp_two_va_one_minus_w);
+    auto log_quotient = log1p(-exp_two_avw) - log1p(-exp_two_av);
+    if (log(w) > log_quotient) {
+      prob += log_diff_exp(log(w), log_quotient);
+      prob = exp(prob);
```

Comment:
Early return for all of these

```suggestion
      return exp(prob);
```

Reply:
Ok. Everywhere changed to early return, where possible.